consensus/downloader.go

```diff
@@ -39,24 +39,20 @@ func newDownloadHelper(c *Consensus, d downloader) *downloadHelper {
 	finishedCh := make(chan struct{}, 1)
 	finishedSub := d.SubscribeDownloadFinished(finishedCh)
 
-	return &downloadHelper{
+	out := &downloadHelper{
 		c:           c,
 		d:           d,
 		startedCh:   startedCh,
 		finishedCh:  finishedCh,
 		startedSub:  startedSub,
 		finishedSub: finishedSub,
 	}
+	go out.downloadStartedLoop()
+	go out.downloadFinishedLoop()
+	return out
 }
 
 func (dh *downloadHelper) start() {
-	go dh.downloadStartedLoop()
-	go dh.downloadFinishedLoop()
```

Comment:
The function `start` initiates an internal loop and awaits the occurrence of `close`, which never happens.